File: admin/src/app/models/view-models/product-detail.ts
import { BaseViewModel,Product } from "@app/models";
import { AutoMap } from "@automapper/classes";

export class ProductDetail extends BaseViewModel {
    @AutoMap()
    id?: number;
    @AutoMap()
    productId: number;
    @AutoMap()
    product: Product;
    @AutoMap()
    price?: number;
    @AutoMap()
    langId: string;
    @AutoMap()
    name: string;
    @AutoMap()
    description: string;
}

File: App.Models/Entities/ProductDetail.cs
using System.ComponentModel.DataAnnotations;
using App.Common.Model;

namespace App.Models.Entities
{
    public class ProductDetail : BaseEntity<int>
    {
        public virtual Product Product { get; set; }
        public int ProductId { get; set; }
        public virtual Lang Lang { get; set; }
        public string LangId { get; set; }
        [MaxLength(256)]
        public string Name { get; set; }
        public string Description { get; set; }
    }
}
